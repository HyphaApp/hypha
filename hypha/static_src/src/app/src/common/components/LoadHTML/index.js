import * as React from "react";
import PropTypes from 'prop-types';


const LoadHTML = props => {
  return <div className="form__group rich-text">
    <div dangerouslySetInnerHTML={{ __html: props.text }} />
  </div>
}

LoadHTML.propTypes={
  text: PropTypes.string
}

LoadHTML.displayName = 'LoadHTML';
export default LoadHTML;
