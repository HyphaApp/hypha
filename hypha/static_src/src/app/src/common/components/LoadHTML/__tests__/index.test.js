import React from "react";
import { mount } from "enzyme";
import sinon from "sinon";
import LoadHTML from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });


describe("Test loadHTML component", () => {
  const text = "test name";

  const subject = mount(<LoadHTML
    text={text}
  />);

  it("Shoud render without issues", () => {
    expect(subject.length).toBe(1);
  });

  it("Should have searched classes ", () => {
    expect(subject.find('.form__group').length).toBe(1);
    expect(subject.children().length).toEqual(1)
  });

  test('render a loadHTML component', () => {
    expect(subject).toMatchSnapshot();
  });

});
