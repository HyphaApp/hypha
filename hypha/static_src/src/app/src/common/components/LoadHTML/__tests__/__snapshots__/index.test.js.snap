// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test loadHTML component render a loadHTML component 1`] = `
<LoadHTML
  text="test name"
>
  <div
    className="form__group rich-text"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "test name",
        }
      }
    />
  </div>
</LoadHTML>
`;
